index.src.html

```diff
@@ -2394,14 +2420,692 @@
+:: The {{ImageTrackList/length}} getter steps are to return the length of
+    {{ImageTrackList/[[track list]]}}.
+
+: <dfn attribute for=ImageTrackList>selectedIndex</dfn>
+:: The {{ImageTrackList/selectedIndex}} getter steps are to return
+    {{ImageTrackList/[[selected index]]}};
+
+: <dfn attribute for=ImageTrackList>onchange</dfn>
+:: An [=event handler IDL attribute=] whose [=event handler event type=] is
+    {{ImageTrackList/change}}.
+
+### Event Summary ### {#imagetracklist-eventsummary}
+
+: <dfn event for=ImageTrackList>change</dfn>
+:: Fired at the {{ImageTrackList}} when the list of {{ImageTrack}}s is
+    established or when attributes of the listed {{ImageTrack}}s are updated.
+
+ImageTrack Interface {#imagetrack-interface}
+--------------------------------------------
+<pre class='idl'>
+<xmp>
+interface ImageTrack {
+  readonly attribute boolean animated;
+  readonly attribute long frameCount;
+  readonly attribute unrestricted float repetitionCount;
+  attribute boolean selected;
+};
+</xmp>
+</pre>
+
+### Internal Slots ### {#imagetrack-internal-slots}
+: <dfn attribute for=ImageTrack>\[[ImageDecoder]]</dfn>
+:: The {{ImageDecoder}} instance that constructed this {{ImageTrack}}.
+
+: <dfn attribute for=ImageTrack>\[[ImageTrackList]]</dfn>
+:: The {{ImageTrackList}} instance that lists this {{ImageTrack}}.
+
+: <dfn attribute for=ImageTrack>\[[animated]]</dfn>
+:: Indicates whether this track contains an animated image with multiple
+    frames.
+
+: <dfn attribute for=ImageTrack>[[frame count]]</dfn>
+:: The number of frames in this track.
+
+: <dfn attribute for=ImageTrack>[[repetition count]]</dfn>
+:: The number of times the animation is intended to repeat.
+
+: <dfn attribute for=ImageTrack>\[[selected]]</dfn>
+:: Indicates whether this track is selected for decoding.
+
+### Attributes ### {#imagetrack-attributes}
+
+: <dfn attribute for=ImageTrack>animated</dfn>
+:: The {{ImageTrack/animated}} getter steps are to return the value of
+    {{ImageTrack/[[animated]]}}.
+
+    NOTE: This attribute provides an early indication that
+        {{ImageTrack/frameCount}} will ultimately exceed 0 for images where the
+        {{ImageTrack/frameCount}} starts at '0' and later increments as new
+        chunks of the {{ReadableStream}} {{ImageDecoderInit/data}} arrive.
+
+: <dfn attribute for=ImageTrack>frameCount</dfn>
+:: The {{ImageTrack/frameCount}} getter steps are to return the value of
+    {{ImageTrack/[[frame count]]}}.
+
+: <dfn attribute for=ImageTrack>repetitionCount</dfn>
+:: The {{ImageTrack/repetitionCount}} getter steps are to return the value of
+    {{ImageTrack/[[repetition count]]}}.
+
+: <dfn attribute for=ImageTrack>selected</dfn>
+:: The {{ImageTrack/selected}} getter steps are to return the value of
+    {{ImageTrack/[[selected]]}}.
+
+    The {{ImageTrack/selected}} setter steps are:
+    1. Let |newValue| be [=the given value=].
+    2. If |newValue| equals {{ImageTrack/[[selected]]}}, abort these steps.
```

Comment:
I think this needs to an abort step if the new value is 'false'. We don't want there to let there be no selected track.

Or is false just supposed to rerun the track selection algorithm?

Reply:
Currently false may cause no tracks to be selected. I think this is possible with Audio|VideoTrackList, so just followed that model. It is an error to call decode() while in that state, checked on step 2 of decode()
```
If [[ImageTrackList]]'s [[selected index]] is '-1', return a Promise rejected with an InvalidStateError DOMException.
```